presidio-analyzer/tests/test_context_support.py

```diff
@@ -90,7 +93,7 @@ def test_text_with_context_improves_score(self):
 
             assert(len(results_without_context) == len(results_with_context))
             for i in range(len(results_with_context)):
-                assert(results_without_context[i].score < results_with_context[i].score)
+                assert(results_without_context[i].score <= results_with_context[i].score)
```

Comment:
You don't get better results with context because the context in the sentence you've provided is too far before the entity location. Please update the example sentence instead of changing the generic test

Reply:
got it, i have reduced the test sentence length as suggested. Also, i am not entirely sure i am running the pytest correctly since the output shown by CI is different to my output. 
Do you mind sharing the correct way to run test?

Reply:
Sure. Here you can find the Python installation instructions: 
https://github.com/microsoft/presidio/blob/master/docs/development.md#setting-up-the-environment---python

Specifically:
```
pipenv run pytest
pipenv run flake8 analyzer --exclude "*pb2*.py"
pipenv run pylint analyzer
```

Reply:
Thanks. Below is the pytest error/warning i am seeing but they are from other recognized. Azure CI does are show the same. What am i missing?
`
_____________________________________________________________ TestSpacyRecognizer.test_person_full_name_complex ______________________________________________________________

self = <tests.test_spacy_recognizer.TestSpacyRecognizer testMethod=test_person_full_name_complex>

    def test_person_full_name_complex(self):
        text = 'Richard (Ric) C. Henderson'
        results = self.prepare_and_analyze(nlp_engine, text)

>       assert len(results) == 3
E       AssertionError: assert 2 == 3
E        +  where 2 = len([<analyzer.recognizer_result.RecognizerResult object at 0x1ac29b350>, <analyzer.recognizer_result.RecognizerResult object at 0x1ac29b690>])

presidio-analyzer/tests/test_spacy_recognizer.py:99: AssertionError
___________________________________________________________ TestSpacyRecognizer.test_person_full_name_with_context ___________________________________________________________

self = <tests.test_spacy_recognizer.TestSpacyRecognizer testMethod=test_person_full_name_with_context>

    def test_person_full_name_with_context(self):
        name = 'John Oliver'
        context = ' is the funniest comedian'
        text = '{} {}'.format(name, context)
        results = self.prepare_and_analyze(nlp_engine, text)

        assert len(results) == 1
        assert_result_within_score_range(
>           results[0], entities[0], 0, 11, NER_STRENGTH, EntityRecognizer.MAX_SCORE)

presidio-analyzer/tests/test_spacy_recognizer.py:53:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
presidio-analyzer/tests/assertions.py:22: in assert_result_within_score_range
    expected_start, expected_end)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

result = <analyzer.recognizer_result.RecognizerResult object at 0x1320fda10>, expected_entity_type = 'PERSON', expected_start = 0, expected_end = 11

    def __assert_result_without_score(result, expected_entity_type,
                                      expected_start, expected_end):
        assert result.entity_type == expected_entity_type
        assert result.start == expected_start
>       assert result.end == expected_end
E       AssertionError

presidio-analyzer/tests/assertions.py:8: AssertionError
============================================================================== warnings summary ==============================================================================
/Users/sunilsattiraju/.pyenv/versions/3.7.4/lib/python3.7/site-packages/google/protobuf/descriptor.py:47
  /Users/sunilsattiraju/.pyenv/versions/3.7.4/lib/python3.7/site-packages/google/protobuf/descriptor.py:47: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from google.protobuf.pyext import _message

/Users/sunilsattiraju/.pyenv/versions/3.7.4/lib/python3.7/site-packages/plac_ext.py:6
  /Users/sunilsattiraju/.pyenv/versions/3.7.4/lib/python3.7/site-packages/plac_ext.py:6: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

-- Docs: https://docs.pytest.org/en/latest/warnings.html
================================================================= 3 failed, 417 passed, 2 warnings in 50.60s =================================================================`

Reply:
Thanks! Probably a version change in spaCy. Let me check

Reply:
We're working on a fix. See #248

Reply:
pipenv run pytest tests passed after the change. Thanks for the fix